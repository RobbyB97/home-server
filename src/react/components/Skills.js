import React from 'react';


export class Skills extends React.Component {
    render() {
        return (
            <section id="Skills" className="skills">
                Skills component
            </section>
        );
    }
}


export default Skills;