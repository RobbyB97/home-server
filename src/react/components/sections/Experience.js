import React from 'react';


export class Experience extends React.Component {
    render() {
        return (
            <section id="Experience" className="experience">
                
            </section>
        );
    }
}


export default Experience;