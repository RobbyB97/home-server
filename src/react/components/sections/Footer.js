import React from 'react';

import Divider from '../Divider'


export class Footer extends React.Component {
    render() {
        return (
            <footer id="Footer" className="footer">
                <Divider section="footer__divider"/>
                
                <div className="footer__content">
                    <div className="footer__links">
                        
                    </div>

                    <p className="footer__copyright">
                        Â© Robert Bergers 2017-2020
                    </p>
                </div>
            </footer>
        );
    }
}


export default Footer;