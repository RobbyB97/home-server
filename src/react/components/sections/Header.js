import React from 'react';

import DesktopNavigation from '../ui/DesktopNavigation';
import MobileNavigation from '../ui/MobileNavigation';
import HomeButton from '../ui/HomeButton';



export class Header extends React.Component {
    render() {
        return (
            <header id="Header" className="header">
                <HomeButton internal />
                <DesktopNavigation />
                <MobileNavigation />
            </header>
        );
    }
}


export default Header;