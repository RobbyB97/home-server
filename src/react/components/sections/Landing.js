import React from 'react';
import Particles from 'react-particles-js';
import ReactModal from 'react-modal'


export class Landing extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            emailModal: false
        };
    }

    /* Controls email modal */
    toggleEmail() {
        this.setState({
            emailModal: !this.state.emailModal
        });
    }
    render() {
        return (
            <section id="Landing" className="landing">

                {/* Background color blocks */}
                <div className="landing__background">
                    <div className="one">
                        <div className="texture">
                            <div></div>
                        </div>
                    </div>
                    <div className="two">
                        <div className="texture">
                            <div></div>
                        </div>
                    </div>
                    <div className="three">
                        
                    </div>
                    <div className="four">
                        <Particles 
                            width="100vw"
                            height="100vh"
                            className="landing__particles"
                            params={{
                                particles: {
                                    number: {
                                        value: 110,
                                        density: {
                                            enable: true,
                                            value_area: 700
                                        }
                                    },
                                    line_linked: {
                                        enable: true,
                                        distance: 200,
                                        color: '#f8f8f8',
                                        opacity: 0.5,
                                        width: 1
                                    },
                                    color: {
                                        value: '#f8f8f8'
                                    }
                                }
                            }}
                        />
                    </div>
                    <div className="five"></div>
                </div>
                    
                {/* Tagline */}
                <section className="landing__tagline">
                    <h2>
                        <span className="left">Robby</span> <span className="right">Bergers</span>
                    </h2>
                    <p className="one">
                        Designer
                    </p>
                    <p className="two">
                        Developer
                    </p>
                </section>

                {/* Menu buttons */}
                <section className="landing__menu">
                    <a className="landing__view" href="#About">
                        View my work
                    </a>
                    
                    <a className="landing__email"
                        onClick={() => {
                            this.toggleEmail();
                        }}
                    >
                        Get in touch
                    </a>
                </section>

                {/* Email modal */}
                <section className={this.state.emailModal ? "landing__emailModal active":"landing__emailModal"}>
                    <p>Email me @ <a href="mailto:bergersr97@gmail.com">bergersr97@gmail.com</a></p>
                </section>

                {/* Link icons */}
                <section className="landing__contact">
                    <a className="landing__github" href="https://www.github.com/robbyb97">
                        <div></div>
                    </a>
                    
                    <a className="landing__linkedin" href="https://www.linkedin.com/in/robbyb97/">
                        <div></div>
                    </a>

                    <a className="landing__upwork" href="https://www.upwork.com/o/profiles/users/~01f696e65011a54866/">
                        <div></div>
                    </a>

                    <a className="landing__twitter" href="https://twitter.com/RobbyBergers">
                        <div></div>
                    </a>
                </section>
            </section>
        );
    }
}


export default Landing;