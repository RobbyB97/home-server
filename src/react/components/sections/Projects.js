import React from 'react';


export class Projects extends React.Component {
    render() {
        return (
            <section id="Projects" className="projects">
                <div className="projects__shape one"></div>
                <div className="projects__shape two"></div>
                Projects component
            </section>
        );
    }
}


export default Projects;