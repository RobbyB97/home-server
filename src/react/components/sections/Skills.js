import React from 'react';

import SkillSection from '../SkillSection';


export class Skills extends React.Component {
    render() {
        const sections = [
            {
                title: "Front end",
                skills: [
                    "React", "SASS", "HTML5", "CSS3", "JS (ES6)"
                ]
            }, {
                title: "Workflow",
                skills: [
                    "Git", "NPM", "Gulp", "Webpack", "Jest", 
                    "Firebase", "Heroku"
                ]
            }, {
                title: "Back end",
                skills: [
                    "Node.js", "BASH", "Linux/Ubuntu server",
                    "MongoDB", "Oracle SQL"
                ]
            }
        ];

        return (
            <section id="Skills" className="skills">
                <header className="skills__header">
                    <h2>My skillset</h2>
                </header>

                {sections.map((section) => (
                    <SkillSection 
                        {...section}
                        key={section.title}
                    />
                ))}
            </section>
        );
    }
}


export default Skills;