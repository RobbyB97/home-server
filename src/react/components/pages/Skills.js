import React from 'react'
import {connect} from 'react-redux'

import Header from '../Header'
import StyleGuide from './StyleGuide'
import Card from '../Card'

import {toggleHomePage} from '../../actions/home'

import {Link, NavLink} from 'react-router-dom'


const mapStateToProps = (state) => {
    return {
      nav: state.navigation.nav,
      title: state.title.title,
      home: state.home.home,
      homePage: state.home.homePage
    }
  }