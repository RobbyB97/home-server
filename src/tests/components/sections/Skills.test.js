import React from 'react'
import {shallow} from 'enzyme'

import {Skills} from '../../../react/components/sections/Skills'


test('Render Skills', () => {
    const wrapper = shallow(<Skills />)
    expect(wrapper).toMatchSnapshot()
})