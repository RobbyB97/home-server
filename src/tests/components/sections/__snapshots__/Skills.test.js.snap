// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Skills 1`] = `
<section
  className="skills"
  id="Skills"
>
  Skills component
</section>
`;
